pkg/autoscaler/autoscaler/calculate/calculate.go

```diff
@@ -46,7 +48,10 @@ type SingleQuery struct {
 func queryMetricsFromPrometheus(tac *v1alpha1.TidbClusterAutoScaler, client promClient.Client, sq *SingleQuery, resp *Response) error {
 	query := sq.Quary
 	timestamp := sq.Timestamp
-	req, err := http.NewRequest("GET", fmt.Sprintf("%s%s", sq.Endpoint, queryPath), nil)
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
```

Comment:
Suggesting defining a variable for `5` here.

Reply:
How about define it as a constant out of this function?

Reply:
Great!